archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterAnnotationsTest.java

```diff
@@ -60,22 +64,29 @@ public void parameters_of_meta_annotations_are_transitively_imported(HasAnnotati
         assertThatAnnotation(metaMetaMetaAnnotation)
                 .hasClassProperty("classParam", SomeMetaMetaMetaAnnotationClassParameter.class)
                 .hasEnumProperty("enumParam", SomeMetaMetaMetaAnnotationEnumParameter.VALUE)
+                .hasEnumDefaultValue("enumParam", SomeMetaMetaMetaAnnotationEnumParameter.CONSTANT)
+                .hasClassDefaultValue("classParam", String.class)
                 .hasAnnotationProperty("annotationParam", SomeMetaMetaMetaParameterAnnotation.class)
                 .withClassProperty("value", SomeMetaMetaMetaParameterAnnotationClassParameter.class);
+
+        assertThatAnnotation(metaMetaMetaAnnotation)
+                .hasAnnotationDefaultValue("annotationParam", SomeMetaMetaMetaParameterAnnotation.class)
+                .withClassProperty("value", Boolean.class);
+
     }
 
     private @interface MetaAnnotationWithParameters {
-        SomeEnum someEnum();
+        SomeEnum someEnum() default SomeEnum.VARIABLE;
```

Comment:
Really good catch here :+1: :clap: I think the process is broken, because we should not mix resolving and completing :thinking: I wish this could be simplified somehow :disappointed:
But if you think about it, for the type hierarchy or call targets, we first resolve, then afterwards we complete everything. I think we should not mix resolving the necessary annotation types and building the annotations together in the same phase :thinking: I'll look into it some more!!

Reply:
Okay, I've tried to split the phases better, can you check again?